src/OMSimulatorLib/Model.cpp

```diff
@@ -1211,3 +1211,34 @@ oms_status_enu_t oms::Model::removeSignalsFromResults(const char* regex)
       return oms_status_error;
   return oms_status_ok;
 }
+
+void oms::Model::exportSignalFilter(pugi::xml_node &node) const
+{
+  if (!system)
+    return;
+
+  std::vector<oms::Connector> filteredSignals;
+  system->getFilteredSignals(filteredSignals);
+
+  for (auto const& signal : filteredSignals)
+  {
+    pugi::xml_node oms_variable = node.append_child(oms::ssp::Version1_0::oms_Variable);
+    oms_variable.append_attribute("name") = signal.getFullName().c_str();
+    oms_variable.append_attribute("type") = signal.getTypeString().c_str();
+    oms_variable.append_attribute("kind") = signal.getCausalityString().c_str();
+  }
+}
+
+oms_status_enu_t oms::Model::importSignalFilter(const std::string& filename, const Snapshot& snapshot)
+{
+  pugi::xml_node oms_signalfilter = snapshot.getResourceNode(filename);
+
+  removeSignalsFromResults(".*"); // disable all signals
+  for (pugi::xml_node_iterator it = oms_signalfilter.begin(); it != oms_signalfilter.end(); ++it)
+  {
+    if (std::string(it->name()) == oms::ssp::Version1_0::oms_Variable)
+      addSignalsToResults(it->attribute("name").as_string());
+  }
+
+  return oms_status_ok;
+}
```

Comment:
We should add all signals in case that the signal filter file doesn't exist.

Reply:
we should handle cases for` signalFilterFilename= ".*" `